packages/@aws-cdk/aws-appmesh/lib/tls-validation.ts

```diff
@@ -0,0 +1,185 @@
+
+  /**
+   * VirtualNode CFN configuration for listener's TLS Validation
+   *
+   * @default - no TLS Validation
+   */
+  readonly virtualNodeListenerTlsValidationTrust?: CfnVirtualNode.ListenerTlsValidationContextTrustProperty
+
+  /**
+   * VirtualGateway CFN configuration for client policy's TLS Validation
+   */
+  readonly virtualGatewayClientTlsValidationTrust: CfnVirtualGateway.VirtualGatewayTlsValidationContextTrustProperty;
+
+  /**
+   * VirtualGateway CFN configuration for listener's TLS Validation
+   *
+   * @default - no TLS Validation
+   */
+  readonly virtualGatewayListenerTlsValidationTrust?: CfnVirtualGateway.VirtualGatewayListenerTlsValidationContextTrustProperty;
+}
+
+/**
+ * ACM Trust Properties
+ */
+export interface TlsValidationAcmTrustOptions {
+  /**
+   * Contains information for your private certificate authority
+   */
+  readonly certificateAuthorities: acmpca.ICertificateAuthority[];
+}
+
+/**
+ * File Trust Properties
+ */
+export interface TlsValidationFileTrustOptions {
+  /**
+   * Path to the Certificate Chain file on the file system where the Envoy is deployed.
+   */
+  readonly certificateChain: string;
+}
+
+/**
+ * Defines the TLS validation context trust.
+ */
+export abstract class TlsValidationTrust {
+  /**
+   * Tells envoy where to fetch the validation context from
+   */
+  public static file(props: TlsValidationFileTrustOptions): TlsValidationTrust {
+    return new TlsValidationFileTrust(props);
+  }
+
+  /**
+   * TLS validation context trust for ACM Private Certificate Authority (CA).
+   */
+  public static acm(props: TlsValidationAcmTrustOptions): TlsValidationTrust {
+    return new TlsValidationAcmTrust(props);
+  }
+
+  /**
+   * Returns Trust context based on trust type.
+   */
+  public abstract bind(scope: Construct): TlsValidationTrustConfig;
+
+}
+
+class TlsValidationAcmTrust extends TlsValidationTrust {
+  /**
+   * Contains information for your private certificate authority
+   */
+  readonly certificateAuthorities: acmpca.ICertificateAuthority[];
+
+  constructor (props: TlsValidationAcmTrustOptions) {
+    super();
+    this.certificateAuthorities = props.certificateAuthorities;
+  }
+
+  public bind(_scope: Construct): TlsValidationTrustConfig {
+    if (this.certificateAuthorities.length === 0) {
+      throw new Error('you must provide at least one Certificate Authority when creating an ACM Trust ClientPolicy');
+    } else {
+      return {
+        virtualNodeClientTlsValidationTrust: {
+          acm: {
+            certificateAuthorityArns: this.certificateAuthorities.map(certificateArn =>
+              certificateArn.certificateAuthorityArn),
+          },
+        },
+        virtualGatewayClientTlsValidationTrust: {
+          acm: {
+            certificateAuthorityArns: this.certificateAuthorities.map(certificateArn =>
+              certificateArn.certificateAuthorityArn),
+          },
+        },
```

Comment:
This is great question.

> Can you help me understand one thing? Why do all 4 of the L1 TlsValidationContextTrustProperty interfaces have the file property, but only 2 of them have the acm property?

This is due to [mTLS feature limitation](https://docs.aws.amazon.com/app-mesh/latest/userguide/mutual-tls.html) that mTLS only supports `file` and `sds` (*this will be added during my subsequent code change) but NOT `acm`.

> Why have 4 properties, if you only ever use 2 of them?

This is very fair question and I think this confusion is brought up since I included the change to add `virtualNodeListenerTlsValidationTrust` and`virtualGatewayListenerTlsValidationTrust`.  These two properties are specific to mTLS feature implementation and **I shouldn't have included them** in this code review. Will remove them.

Reply:
Hmm. So how will you model this in the CDK, so that places that don't allow passing `acm` in the service do not allow passing `TlsValidationTrust.acm()` in the CDK?

Ideally, this would be a mistake that we prevent at compile-time.

Reply:
That is another great question. We had a team discussion on whether to enforce the compile-time error here and we concluded not to.
Our reasoning was that `acm` could be supported in future and we would like to avoid submitting the CDK change when that happens.

Reply:
Is there a specific plan to enable the ACM option where it's currently not allowed?

Because if it's only a "maybe this will be allowed one day", I really don't like this sort of future-proofing.
And in that case, I would at least add runtime validation that checks whether `TlsValidationTrust.acm()` is passed somewhere it's not expected.

Reply:
Actually now that I think of this, I think users would get compile-time error if they attempt to pass `TlsValidationTrust.acm()` where it is not supporting `acm`.

For example, if a user attempt to pass `TlsValidationTrust` to `TlsListener`, we should expect the compile-time error because `ListenerTlsValidationContextTrust` does not support [`acm`](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-appmesh-virtualnode-listenertlsvalidationcontexttrust.html).

Reply:
I'm not following...

Both `TlsValidationTrust.file()` and `TlsValidationTrust.acm()` have the same return type (`TlsValidationTrust`).

How is it possible that you can pass `TlsValidationTrust.file()` somewhere, but passing `TlsValidationTrust.acm()` in that same place would not compile?

Reply:
You are right on this. I went ahead and implemented the change to try inserting `TlsValidationTrust.acm()` but I did not get both compile-time and run-time error. 

As your suggested earlier, I will add run-time to check.

Reply:
Sounds good!